frontend/src/lib/templates/page/Page.svelte

```diff
@@ -0,0 +1,195 @@
+  export let progressTitle: $$Props['progressTitle'] = $_('header.progressTitle');
+
+  let drawerOpen = false;
+  let drawerCloseElement: HTMLButtonElement | undefined = undefined;
+  let drawerOpenElement: HTMLButtonElement | undefined = undefined;
+
+  /**
+   * Toggle the drawer between open and closed. We also focus on the
+   * relevant element to make it easy to toggle it back when using
+   * keyboard navigation.
+   */
+  function toggleDrawer() {
+    drawerOpen = !drawerOpen;
+    const el = drawerOpen ? drawerCloseElement : drawerOpenElement;
+    el?.focus();
+  }
+
+  // Merge the necessary drawer class to possible extra classes defined in the props
+  $$restProps.class = `drawer ${$$restProps.class ?? ''}`;
+
+  // TODO: Fix progress bar color on iOS Safari:
+  // see https://stackoverflow.com/questions/38622911/styling-meter-bar-for-mozilla-and-safari
+  // TODO: Consider whether we could allow passing of any HTMLElement props to the
+  // container elements of the slots instead of just class and label for nav.
+  // TODO: We might need to convert the checkbox that controls the menu to a
+  // button and handle showing/hiding with on:click in order to use proper Aria
+  // attributes: see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded
+</script>
+
+<!--
+@component
+Use as an abstract template for creating page templates, which still share the 
+most important parts of the app layout. It is based on DaisyUI's 
+[Drawer component](https://daisyui.com/components/drawer/).
+
+The content is provided in named slots with the default slot contents forming
+the page's basic content.
+
+Use the properties to add to the classes of the elements containing the slots,
+to define some ids, pass Aria labels and show an optional progress bar in the
+header. You can also pass any valid attributes to the root `<div>` element of 
+the Drawer component.
+
+### Properties
+
+- `title`: The required title of the page.
+- See `Page.type.ts` or code completion info for the other properties.
+
+### Slots
+
+- default: the page's basic content
+- `header`: content for the rest of the `<header>` after `navOpen`
+- `navOpen`: icon or text for the button that opens the menu. @default The app
+  logo and the menu icon
+- `navClose`: icon or text for the button that closes the menu from within it. 
+  @default A close icon
+
+### Usage
+
+```tsx
+  <Page title="Example page" asideClass="text-secondary text-center">
+    <svelte:fragment slot="header">
+      <Button on:click={showInfo} variant="icon" icon="info" text="Show info"/>
+    </svelte:fragment>
+    <h1>The page title</h1>
+    <p>Main body text</p>
+  </Page>
+```
+-->
+
+<!-- Page title -->
+<svelte:head>
+  <title>{title} – {$page.data.appLabels.appTitle}</title>
+</svelte:head>
+
+<!-- Skip links for screen readers and keyboard users -->
+<a href="#{contentId}" class="sr-only focus:not-sr-only">{skipLinkLabel}</a>
+
+<!-- Drawer container -->
```

Comment:
That's strange, bc on my Chrome the tab key works as supposed, but it seems that we should make this work using the arrow keys, as you suggested earlier. I'll try and fix that, but then there's the question should the menu, once open, be modal bc – or at least close automatically if the tab key is pressed… It's surprisingly involved 😅

Reply:
Okay, now I updated this, which turned out be quite complicated with encountering some Svelte bugs as well.

Can you test the keyboard navigation now on your side? The main changes are:

1. When the drawer is closed, the whole navigation has `class="hidden"`, which entails `display: none`
2. When the drawer is open, the rest of the page has `inert="true"`, which renders it inaccessible by keyboard, in essence making the menu modal*
3. The button that closes the navigation is now inside the `<nav>` element produced by `Navigation`. This seems like a more logical place for it, because interaction can now nicely be limited to within the `<nav>`.

*The browser support for the pure html attribute is ~ 91%. There's a [polyfill](https://github.com/WICG/inert) available as well, which we might consider.